### DISCUSSION TITLE
Duckdb dict cursor

### DISCUSSION THREAD

**Discussion by arunksoman** [duckdb/duckdb#9781] [General]
Does duckdb support dict cursor? Currently fetchall() returns list of tuples? Can we have list of dicts instead.

**Comment by jackbravo** [comment]
You can call fetch_df instead, and get a pandas DataFrame that allows access by column name and other features.

**Reply by arunksoman** [reply]
@jackbravo Consider I need to query data from a Postgres database for a REST API; I believe it is not a good idea to convert the data to a Pandas dataframe and convert later to a dictionary. If the output already is a dictionary, it would be easy to convert things to a response model without additional operations.

**Comment by Tishj** [comment]
It would be great if you could give an example query and the way you would want this represented as a dict when the result is fetched, we could consider adding a `fetchdict` method

**Reply by jackbravo** [reply]
For big resultsets it would be great. And something similar to what sqlaclhemy does, of returning [named tuples](https://docs.sqlalchemy.org/en/20/core/connections.html#sqlalchemy.engine.Row) would be ideal, IMO.

That way you can access either by column number or name on each row.
